product_supplierinfo_revision/wizards/supplierinfo_duplicate_wizard.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2017 Carlos Dauden <carlos.dauden@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from dateutil.relativedelta import relativedelta
+
+from odoo import fields, models
+
+
+class ProductSupplierInfoDuplicateWizard(models.TransientModel):
+    _name = "product.supplierinfo.duplicate.wizard"
+    _description = "Product Supplier Duplicate Wizard"
+
+    date_start = fields.Date(required=True)
+    date_end = fields.Date()
+    variation_percent = fields.Float(digits="Product Price", string="Variation %",)
+
+    def action_apply(self):
+        Supplierinfo = self.env["product.supplierinfo"]
```

Comment:
You shouldn't use upper case characters to start a variable in Python
```suggestion
        supplierinfo = self.env["product.supplierinfo"]
```